src/gridfs-stream/upload.ts

```diff
@@ -151,48 +177,53 @@ export class GridFSBucketWriteStream extends Writable {
    * @param {GridFSBucket~errorCallback} callback Function to call when all files and chunks have been persisted to MongoDB
    */
 
-  end(chunk: any, encoding?: any, callback?: Callback) {
-    var _this = this;
-    if (typeof chunk === 'function') {
-      (callback = chunk), (chunk = null), (encoding = null);
-    } else if (typeof encoding === 'function') {
-      (callback = encoding), (encoding = null);
-    }
+  end(
+    chunkOrCallback?: Buffer | Callback<GridFsFile | undefined>,
```

Comment:
`Callback<GridFsFile>` is equivalent to `Callback<GridFsFile | undefined>`

Reply:
Updated to `void` but still needs this union for compatibility with `checkAborted`.